rs/registry/canister/src/mutations/node_management/do_add_node.rs

```diff
@@ -145,6 +160,18 @@ impl Registry {
     }
 }
 
+// try to convert input string inot NodeType enum
+fn try_str_to_node_type(type_string: &str) -> Result<NodeType, String> {
+    Ok(match type_string {
```

Comment:
@sasa-tomic This is where you could stop new nodes from being registered with old types, by matching on strings and then returning particular error messages.